RHIDP-11676: Orchestrator - Migrate doc for Log aggregation and Open Telemetry feature (#2027)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/proc-collect-traces-to-see-execution-timing-and-failures.adoc

:_mod-docs-content-type: PROCEDURE

[id="collect-traces-to-see-execution-timing-and-failures_{context}"]

> = Collect traces to see execution timing and failures

Reviewer: ```suggestion
= Collect traces to monitor workflow performance
```
(Would this title work??)

[role="_abstract"]

> Turn on distributed tracing to watch workflows execute step-by-step across services. Traces show exactly which operations run, how long they take, and where failures occur in the execution path.

Reviewer: ```suggestion
To monitor workflows in {product}, you must enable distributed tracing. Distributed tracing shows the executed path of workflow steps and identifies where failures occur. 
```

# OpenTelemetry Configuration
quarkus.otel.enabled=true
quarkus.otel.traces.enabled=true
quarkus.otel.metrics.enabled=true
quarkus.otel.logs.enabled=true

# Service Resource Attributes
quarkus.otel.resource.attributes=\
  service.name=my-workflow,\
  service.namespace=workflows,\
  service.version=1.0.0,\
  deployment.environment=production

# SonataFlow Specific Configuration
# Master switch for SonataFlow OpenTelemetry integration
sonataflow.otel.enabled=true
# Service identification (uses Quarkus application name/version as defaults)
sonataflow.otel.service-name=${quarkus.application.name:kogito-workflow-service}
sonataflow.otel.service-version=${quarkus.application.version:unknown}
# Enable span creation for workflow states
sonataflow.otel.spans.enabled=true
# Enable process lifecycle events (start, complete, error, state transitions)
sonataflow.otel.events.enabled=true
----

. Save the ConfigMap and restart the workflow pod to apply the changes.

.Verification

. Verify that the OpenTelemetry addon is loaded by checking the pod logs:
+
[source,bash]
----
oc logs -n workflows deployment/onboarding-workflow | grep "sonataflow-addons-quarkus-opentelemetry"
----

> . Verify successful trace export:

Reviewer: ```suggestion
. Verify the trace report status:
```

+
[source,bash]
----
oc logs -n workflows deployment/greeting | grep -i "export\|batch"
----

> . Confirm that the observability backend (for example, Jaeger) is receiving data:

Reviewer: ```suggestion
. Confirm that the observability backend, such as `Jaeger`, is receiving data:
```
